### ISSUE TITLE
Additional data get truncated

### ISSUE THREAD

**Issue by stephen1706** [getsentry/sentry-java#863]
Hi, I am trying to put a long string of additional data by calling `event.withExtra(key, value)` to the SDK. However, I found that my string value got truncated after 397 characters long in the Sentry dashboard. Is this a limitation on Android SDK? Because in iOS, it doesn't get truncated.
Thank you

**Comment by marandaneto** [comment]
ok, we'll check this out.
do you have a link to the issue? could you check the raw JSON file size to see if the file size limits are the issue?

meanwhile, I invite you to try out sentry-android v2 :)

https://github.com/getsentry/sentry-android

**Comment by stephen1706** [comment]
I don't think I can share the link to the issue now. The raw JSON file is only 2kb, so shouldn't be a problem.

**Comment by marandaneto** [comment]
@stephen1706 thanks for the link.

so the SDK defaults the trimming to 1000 chars, have you overwritten this config? https://docs.sentry.io/clients/java/config/#max-message-size

**Comment by marandaneto** [comment]
ok found out there's a hard limit of 400 chars and there's no easy way to change that, ideally, we'd offer an option as we do with `maxMessageSize` to config the `maxStringSize`.

You'd need to create an instance of `JsonMarshaller` and `SentryJsonGenerator` overwriting the `maxLengthString` field.

I'll consider that as an improvement, to expose an API to change that on a high-level API instead of doing all of this.

As you are using Android, I'd suggest that you migrate to the Android SDK v2, which is sitting here https://github.com/getsentry/sentry-android

We don't have such limits on the v2, but only payload size which is described on the link shared above (my 1st message).

Thanks for raising this :)

**Comment by apiening** [comment]
I'm using **Sentry 1.7.30** with **Java** (Linux server, not android) and have the same issue: I'm adding the SQL statement that failed with `Sentry.getContext().addExtra("SQL", lastSQLstatement);` to the Sentry context before capturing the exception.
The SQL statement gets transferred already truncated in the JSON.
Since the SQL statement is truncated, there is no way to analyze the root cause of the error.

Please can you provide us with a workaround or even better an updated release where the hard limit is set to something like 4096 chars?

**Comment by marandaneto** [comment]
@apiening the message above has a workaround.

> You'd need to create an instance of JsonMarshaller and SentryJsonGenerator overwriting the maxLengthString field

basically you'd need to create your own instance of SentryClient with those overwritten dependencies.

**Comment by marandaneto** [comment]
https://github.com/getsentry/sentry-java/releases/tag/3.0.0-beta.1
size limit now is only on the server

tldr
https://docs.sentry.io/platforms/rust/enriching-events/context/
200kb
